src/commonMain/kotlin/fr/acinq/lightning/channel/states/Channel.kt

```diff
@@ -501,18 +506,48 @@ sealed class ChannelStateWithCommitments : PersistedChannelState() {
                     })
                 }
                 else -> {
-                    logger.warning { "unrecognized tx=${tx.txid}" }
-                    // This can happen if the user has two devices.
-                    // - user creates a wallet on device #1
-                    // - user restores the same wallet on device #2
-                    // - user does a splice on device #2
-                    // - user starts wallet on device #1
-                    // The wallet on device #1 has a previous version of the channel, it is not aware of the splice tx. It won't be able
-                    // to recognize the tx when the watcher notifies that the (old) funding tx was spent.
-                    // However, there is a race with the reconnection logic, because then the device #1 will recover its latest state from the
-                    // remote backup.
-                    // So, the best thing to do here is to ignore the spending tx.
-                    Pair(this@ChannelStateWithCommitments, listOf())
+                    // Our peer may publish an alternative version of their commitment using a different feerate.
+                    val alternativeFeerateCommits = buildList {
+                        add(commitments.latest.remoteCommit)
+                        commitments.latest.nextRemoteCommit?.let { add(it.commit) }
+                    }.filter { remoteCommit ->
+                        remoteCommit.spec.htlcs.isEmpty()
+                    }.flatMap { remoteCommit ->
+                        Commitments.alternativeFeerates.map { feerate ->
+                            val alternativeSpec = remoteCommit.spec.copy(feerate = feerate)
+                            val (alternativeRemoteCommitTx, _) = Commitments.makeRemoteTxs(channelKeys(), remoteCommit.index, commitments.params.localParams, commitments.params.remoteParams, commitments.latest.fundingTxIndex, commitments.latest.remoteFundingPubkey, commitments.latest.commitInput, remoteCommit.remotePerCommitmentPoint, alternativeSpec)
+                            RemoteCommit(remoteCommit.index, alternativeSpec, alternativeRemoteCommitTx.tx.txid, remoteCommit.remotePerCommitmentPoint)
+                        }
+                    }
```

Comment:
Could we move this to `Commitments.alternativeFeerateCommits()` right under `Commitments.alternativeFeerates` (which could be made private)?

This would simply a bit this pretty scary `handleRemoteSpentOther()` function and contain the alternate sigs stuff in one place.

Reply:
Agreed, done in https://github.com/ACINQ/lightning-kmp/pull/553/commits/abee6c6eb1fd120d49c653829380478973bf2ce7
I didn't make `alternativeFeerates` private though, because that would require moving the alternative sig generation in a helper function, and I think it's clearer to have it inline in `sendCommit` right below the "normal" signature.